source/tutorials/learning-journey/packaging-existing-software.md

```diff
@@ -0,0 +1,531 @@
+---
+myst:
+  html_meta:
+    "description lang=en": "Packaging Existing Software With Nix"
+    "keywords": "Nix, packaging"
+---
+
+
+(packaging-existing-software)=
+
+
+# Packaging Existing Software With Nix
+
+One of Nix's primary use-cases is in addressing common difficulties encountered while packaging software, like *managing dependencies*.
+
+In the long term, Nix helps tremendously in alleviating that stress, but when *first* (re)packaging existing software with Nix, it's common to encounter missing dependencies preventing builds from succeeding.
+
+In this tutorial, you'll create your first Nix derivations to package C/C++ software, taking advantage of the [`nixpkgs` `stdenv`](https://nixos.org/manual/nixpkgs/stable/#chap-stdenv) which automates much of the work of building self-contained C/C++ packages.
+
+The tutorial begins by considering `hello`, an implementation of "hello world" which only requires dependencies already in `stdenv`.
+
+Next, you will build more complex packages with their own dependencies, leading you to use additional derivation features.
+
+You'll encounter and address Nix error messages, build failures, and a host of other issues, developing your iterative debugging techniques along the way.
+
+:::{note}
+An important point of clarification: the term "package" is used conventionally by analogy to other systems, although the term does not refer to a proper concept in Nix.
```

Comment:
So Nix doesn't have a package concept, but Nixpkgs definitely does, it's just not formally defined. I think this would be a concise and accurate description:

```suggestion
A _package_ is an informally defined Nixpkgs concept referring to a Nix derivation representing an installation of some project.
Packages have mostly standardised attributes and output layouts, allowing them to be discovered in searches and installed into environments alongside other packages.
```